x-pack/elastic-agent/pkg/core/plugin/process/app.go

```diff
@@ -192,16 +196,32 @@ func (a *Application) watch(ctx context.Context, p app.Taggable, proc *process.I
 		case ps := <-a.waitProc(proc.Process):
 			procState = ps
 		case <-a.bgContext.Done():
-			a.Stop()
+			return
```

Comment:
Just curious: is there any connection between `a.bgContext` and `ctx`? Does `ctx` has to be cancelled when `a.bgContext` is cancelled?

Reply:
bgContext is like a cancellation token passed from top of the app and cancelled on exit or unenroll, so agent cleans and backs up everything in a nice manner. we want to avoid just shutting down agent without any cleaning as this may turn out problematic